packages/spanner/src/transaction-request.js

```diff
@@ -51,7 +51,14 @@ var PartialResultStream = require('./partial-result-stream.js');
  * @param {object=} options - Timestamp options.
  */
 function TransactionRequest(options) {
+  this.readOnly = false;
+
   if (options && !is.empty(options)) {
+    options = extend({}, options);
+
+    this.readOnly = !!options.readOnly;
+    delete options.readOnly;
```

Comment:
why do you need to delete?

Reply:
We send this object when creating the transaction, if grpc finds a key that is not listed within a proto file it will throw an error.